cdap-app-fabric/src/main/java/io/cdap/cdap/internal/app/runtime/ProgramRunners.java

```diff
@@ -186,6 +190,21 @@ public static ClusterMode getClusterMode(ProgramOptions programOptions) {
     return clusterMode == null ? ClusterMode.ON_PREMISE : ClusterMode.valueOf(clusterMode);
   }
 
+  /**
+   * Create a program context given a {@link ProgramRunId}.
+   * @param programRunId the {@link ProgramRunId} of the current execution
+   * @return the program context
+   */
+
+  public static Map<String, String> createProgramContext(ProgramRunId programRunId) {
```

Comment:
ProgramContext refer to something else. Why not have this method takes the `MetricsCollectionService` as well and return ` MetricsContext` and call this method `createProgramMetricsContext`?

Reply:
fixed